### PR TITLE
Respect package versions in simulate modes

### PR DESCRIPTION
**Pull request by vpodzime**

### CODE REVIEW

**Review by larsewi — APPROVED** [review]
Great work!

File: cf-agent/simulate_mode.c [hunk 1]
```diff
@@ -697,6 +697,64 @@ static void PkgOperationRecordDestroy(PkgOperationRecord *pkg_op)
...
+    assert(!NULL_OR_EMPTY(ver1) && !NULL_OR_EMPTY(ver2));
+
+    /* "latest" is greater than any version */
+    if (StringEqual(ver1, "latest"))
+    {
+        return true;
+    }
+
+    /* TODO: do real version comparison */
+    return (StringSafeCompare(ver1, ver2) == 1);
```
Comment (olehermanse) [inline cf-agent/simulate_mode.c:713]: https://github.com/cfengine/libntech/commit/eff7196d26a841af40fbd8b36e1b6064bceb3e28

Could reuse / improve that. It expects 1-3 numbers separated by dots, and nothing else, currently.

**Review by olehermanse — APPROVED** [review]

File: cf-agent/simulate_mode.c (hunk 1 above)
Comment (vpodzime) [reply cf-agent/simulate_mode.c:713]: Yup.

**Comment by vpodzime** [comment]
@cf-bottom jenkins

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/6227/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/6227/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-6227/

**Pull request merged**